src/resources/js/tickets-block.js

```diff
@@ -794,9 +582,904 @@ tribe.tickets.block = {
+			ticket.qty   = 0;
+			ticket.price = obj.getPrice( $cartItem );
+
+			obj.updateTotal( ticket.qty, ticket.price, $cartItem );
+			obj.updateFormTotals( $cart );
+
+			$( '.tribe-tickets__item__attendee__fields__container[data-ticket-id="' + ticket.id + '"]' )
+				.removeClass( 'tribe-tickets--has-tickets' )
+				.find( obj.modalSelector.arItem ).remove();
+		}
+	);
+
+	/**
+	 * Adds focus effect to ticket block.
+	 *
+	 * @since TBD
+	 *
+	 */
+	obj.document.on(
+		'focus',
+		'.tribe-ticket .ticket-meta',
+		function( e ) {
+			var input      = e.target;
+			obj.focusTicketBlock( input );
+		}
+	);
+
+	/**
+	 * handles input blur.
+	 *
+	 * @since TBD
+	 *
+	 */
+	obj.document.on(
+		'blur',
+		'.tribe-ticket .ticket-meta',
+		function( e ) {
+			var input      = e.target;
+			obj.unfocusTicketBlock( input );
+
+			obj.checkIfFieldValid( input );
+
+			var $form     = $( input ).closest( 'form' );
+			obj.validateForm( $form );
+		}
+	);
+
+	/**
+	 * Handle the Ticket form(s).
+	 *
+	 * @since 4.9
+	 *
+	 * @return void
+	 */
+	obj.document.on(
+		'change keyup',
+		obj.selector.itemQuantityInput,
+		function( e ) {
+			var $this        = $( this );
+			var $ticket      = $this.closest( obj.selector.item );
+			var $ticket_id   = $ticket.data( 'ticket-id' );
+			var $form        = $this.closest( 'form' );
+			var new_quantity = parseInt( $this.val(), 10 );
+			new_quantity     = isNaN( new_quantity ) ? 0 : new_quantity;
+
+			e.preventDefault();
+			obj.maybeShowOptOut( $ticket, new_quantity );
+			obj.updateFooter( $form );
+			obj.updateFormTotals( $form );
+
+			// Only disable / enable if is a Tribe Commerce Paypal form.
+			if ( 'Tribe__Tickets__Commerce__PayPal__Main' === $form.data( 'provider' ) ) {
+				obj.tribeCommerceDisable( new_quantity, $form, $ticket_id );
+			}
+		}
+	);
+
+	/**
+	 * Stores to sessionStorage onbeforeunload for accidental refreshes, etc.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	obj.document.on(
+		'beforeunload',
+		function( e ) {
+			if ( tribe.tickets.modal_redirect ) {
+				return;
+			}
+
+			obj.storeLocal();
+		}
+	);
+
+	/**
+	 * Handle Modal submission.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	obj.document.on(
+		'click',
+		obj.modalSelector.submit,
+		function( e ) {
+			e.preventDefault();
+			var $button    = $( this );
+			var post_id  = $( '.status-publish' ).attr( 'id' ).replace( 'post-', '' );
+			var provider = $tribe_ticket.data( 'provider' );
+
+			// save meta and cart
+			var params = {
+				provider: obj.commerceSelector[provider],
+				tickets : obj.getTicketsForCart(),
+				meta    : obj.getMetaForSave(),
+			};
+
+			$.ajax({
+				type: 'POST',
+				url: '/wp-json/tribe/tickets/v1/cart/' + post_id,
+				data: params,
+				success: function( response ) {
+					//redirect
+					var url = response.checkout_url;
+
+					if( 'cart-button' === $button.attr( 'name' ) ) {
+						url = response.cart_url
+					}
+
+					if ( 'woo' === obj.commerceSelector[provider]) {
```

Comment:
Shouldn't this be handled by the REST API response with the cart/checkout URLs?

Reply:
And should it apply to cart AND checkout URLs? Asking since I've got the code open for the REST API and I can make this change there

Reply:
I'm not sure - it was something I was trying to get through to checkout. It didn't work so let me just remove it for now.

Reply:
👍